client/app/bundles/course/assessment/submission/containers/ImportedFileView.jsx

```diff
@@ -112,11 +112,11 @@ VisibleImportedFileView.propTypes = {
 };
 
 function mapStateToProps(state, ownProps) {
-  const { displayFileIndex, handleFileTabbing, handleDeleteFile, files } = ownProps;
+  const { displayFileIndex, handleFileTabbing, handleDeleteFile, files, viewHistory } = ownProps;
   const { submission } = state;
   const canDestroyFiles =
     submission.workflowState === workflowStates.Attempting &&
-    submission.isCreator;
+    submission.isCreator && !viewHistory;
```

Comment:
Can we also make sure that the backend ability will handle this, just in case a request to destroy a file was sent?

Reply:
Added in the programming answer's `programming_ability`. Checks that can only delete/create files if it is the `current_answer`